cli/src/cmd/run.rs

```diff
@@ -207,16 +238,64 @@ impl Cmd for RunArgs {
     }
 }
 
+struct ExtraLinkingInfo<'a> {
+    no_target_name: bool,
+    target_fname: String,
+    contract: &'a mut CompactContractBytecode,
+    dependencies: &'a mut Vec<ethers::types::Bytes>,
+}
+
+fn link_key_construction(file: String, key: String) -> (String, String, String) {
+    (file.to_string() + ":" + &key, file, key.to_string())
+}
+
+// post link step for run command
+fn post_link(
```

Comment:
seems like this could be a struct function?

Reply:
since we pass in a fn ptr, this doesnt really work